.github/workflows/code-review.yml

```
@@ -0,0 +1,38 @@
+name: AI Code Review
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    paths:
+      - '**.py'
+      - '**.yml'
+      - '**.md'
+
+permissions:
+  pull-requests: write
+  contents: read
+  checks: write
+
+jobs:
+  review:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install --upgrade PyGithub google-genai requests python-dotenv
+
```

It might be helpful to log the versions of the installed packages for debugging purposes.

```suggestion
|
``````yaml
run: |
python -m pip install --upgrade pip
pip install --upgrade PyGithub google-genai requests python-dotenv
pip freeze > requirements.txt
```